packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2676,4 +2682,25 @@ export class TransactionController extends BaseControllerV1<
       log('Error while updating post transaction balance', error);
     }
   }
+
+  private async publishTransactionWithValidation(
+    rawTx: string,
+    transactionMeta: TransactionMeta,
+  ): Promise<string> {
+    let hash;
+    try {
+      hash = await this.publishTransaction(rawTx);
+      // TODO: Replace `any` with type
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      if (error?.code === errorCodes.rpc.invalidInput) {
+        await this.pendingTransactionTracker.isTransactionConfirmed(
+          transactionMeta,
+        );
+        throw new Error('Transaction already confirmed.');
```

Comment:
Minor, and pending review from Barbara but no periods in single line messages.

Reply:
Fixed in https://github.com/MetaMask/core/pull/3800/commits/e0980caae73e868e350f724def200f65f83811d8